utils/server/tiktoken.ts

```diff
@@ -21,19 +22,45 @@ export async function prepareMessagesToSend(
   modelId: OpenAIModelID
 ): Promise<Message[]> {
   const encoding = await getTiktokenEncoding()
+  const [messagesToSend, requiredTokens] = reduceMessagesToSend(
+    tokenLimit,
+    maxReplyTokens,
+    prompt,
+    messages,
+    modelId,
+    encoding
+  )
+
+  if (requiredTokens > tokenLimit) {
```

Comment:
See Slack discussion. Not sure if this is sufficiently user friendly. It's code-simple, but user unfriendly. But maybe we should leave it like this and add an auto-summarize and restart conversation instead (see issue https://github.com/rijnb/chatty-server/issues/53).

Reply:
this is kind of a safety net, when even system prompt + user prompt (first message) don't fit.
skipping messages still happening [reduceMessagesToSend](https://github.com/rijnb/chatty-server/pull/76/files#diff-0a44d093b1e2a00d0b358a8e7de28b0594346aeef6c6fcd37806a21a5583c5e9R41-R43)